packages/api/endpoints/distribution.js

```diff
@@ -284,7 +291,53 @@ async function handleFileRequest(req, res) {
     .send('Redirecting');
 }
 
+/**
+ * Takes a bucketlist and a bucket/key event, gets the mapping path for each bucket from bucket map,
+ * and writes a bucket mapping object to S3.   Returns the bucket map object.
+ *
+ * @param {Object} event              - Event containing
+ * @param {string[]} event.bucketList - An array of buckets to cache values for
+ * @param {string} event.s3Bucket     - Bucket to write .json map cache file to
+ * @param {string} event.s3Key        - Key to write .json map cache file to
+ * @returns {Promise<Object>}         - A bucketmap object {bucket1: mapping1, bucket2: mapping2}
+ */
+async function handleCreateBucketMapCacheRequest({
+  bucketList,
+  s3Bucket,
+  s3Key,
+}) {
+  if (!bucketList || !s3Bucket || !s3Key) {
+    throw new Error('A bucketlist and s3 bucket/key must be provided in the event');
+  }
+
+  const bucketMap = await getBucketMap();
+
+  const bucketMapObjects = bucketList.map((bucket) => {
```

Comment:
Nit: Some of this logic is somewhat duplicative of the TeaMapCache code.... looking at it it's not clear to me it'd be worth the effort to DRY it up and/or consider moving it into a lib/package/?, but something to consider.

Reply:
They are similar, but are calling different methods `await getTeaBucketPath` vs `await getBucketMap` and `getPathsByPrefixedBucketName`

Reply:
*nods* no worries, it's entirely a nit || take/leave thing.